### PR TITLE
Mongodb backend done

### PR DESCRIPTION
**Pull request by AnkitaBagale**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,65 +1,91 @@
...
-        });
-      } catch {}
+        const {
+          data: { response }
+        } = await getProductsFromServer(
+          "https://uandistoreapi.herokuapp.com/products"
+        );
+        dispatch({ type: "SET_PRODUCTS", payload: response });
+      } catch (error) {
+        console.log(error);
```
Comment (payalkherajani) [inline src/App.js:41]: remove console from here. You can use here notification (toastify) to send error to user

File: src/App.js [hunk 2]
```diff
@@ -1,65 +1,91 @@
...
+        try {
+          const {
+            data: { response }
+          } = await getProductsFromServer(
+            `https://uandistoreapi.herokuapp.com/carts/${userId}/cart`
+          );
+
+          dispatch({ type: "SET_CART", payload: response });
+        } catch (error) {
+          console.log(error);
```
Comment (payalkherajani) [inline src/App.js:58]: same

File: src/App.js [hunk 3]
```diff
@@ -1,65 +1,91 @@
...
+          const {
+            data: { response }
+          } = await getProductsFromServer(
+            `https://uandistoreapi.herokuapp.com/wishlists/${userId}/wishlist`
+          );
+
+          dispatch({ type: "SET_WISHLIST", payload: response });
+        } catch (error) {
+          console.log(error);
+        }
```
Comment (payalkherajani) [inline src/App.js:71]: same here, console

File: src/App.js [hunk 4]
```diff
@@ -1,65 +1,91 @@
...
       <div className="App-container">
         <Nav />
         <Routes>
           <Route path="/" element={<Home />} />
           <Route path="shop" element={<ProductListing />} />
+          <Route path="/shop/:id" element={<ProductDetailPage />} />
+
           <PrivateRoute path="wishlist" element={<Wishlist />} />
           <PrivateRoute path="cart" element={<Cart />} />
           <Route path="/login" element={<Login />} />
```
Comment (payalkherajani) [inline src/App.js]: try putting / in front of path or maintain a consistency. Either apply / in all or in no route

File: src/Components/Authentication/ForgotPasswordPage.jsx [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+
+  const onFocusClearError = (type) => {
+    errorsDispatch({ type, payload: "" });
+  };
+
+  const checkFormValidity = () => {
+    let errorFlag = true;
+
+    if (formState.username === "" || !/^.+@.+\.com$/.test(formState.username)) {
+      errorsDispatch({
```
Comment (payalkherajani) [inline src/Components/Authentication/ForgotPasswordPage.jsx]: seems like a email check, but then why variable name is username ? Can you please rename it to email for proper understanding!

File: src/Components/Authentication/ForgotPasswordPage.jsx [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+      });
+      if (response.status === 200) {
+        setLoading(false);
+        setSuccess(true);
+      } else {
+        setLoading(false);
+        setError(
+          response?.data?.message || "Something went wrong, please try again!"
+        );
+      }
```
Comment (payalkherajani) [inline src/Components/Authentication/ForgotPasswordPage.jsx]: Please here, use try catch

File: src/Components/Authentication/ForgotPasswordPage.jsx [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+                }}
+              />
+              <div
+                style={{
+                  display:
+                    fieldErrors.confirmPasswordError ||
+                    formState.password !== formState.confirmPassword
+                      ? "block"
+                      : "none"
+                }}
```
Comment (payalkherajani) [inline src/Components/Authentication/ForgotPasswordPage.jsx:216]: Please remove inline-styling

File: src/Components/Authentication/ForgotPasswordPage.jsx [hunk 8]
```diff
@@ -0,0 +1,265 @@
...
+              onClick={formSubmitHandler}
+            >
+              RESET
+            </button>
+          </form>
+          <div
+            style={{
+              display: error !== "" ? "block" : "none",
+              color: "#ff9204"
+            }}
```
Comment (payalkherajani) [inline src/Components/Authentication/ForgotPasswordPage.jsx:241]: Please here also, remove inline styling

File: src/Components/Authentication/Login.jsx [hunk 9]
```diff
@@ -35,13 +36,14 @@ export const Login = () => {
...
       password: formState.password,
       from: state?.from ? state.from : "/"
     });
-    if (res.status === 401) {
-      setError(res.response);
+
+    if (res?.status === 401 || res?.status === 500) {
+      setError(res?.data?.message || "Something went wrong, please try again!");
       setLoading(false);
     }
```
Comment (payalkherajani) [inline src/Components/Authentication/Login.jsx]: Can we use here try catch ? If yes , please change this

File: src/Components/ErrorPage/ErrorPage.jsx [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+import pencilgif from "./Images/earaser2.gif";
+import "./error-page.css";
+
+export const ErrorPage = () => {
+  return (
+    <>
+      <div className="error-message-container">
+        <div className="text-center">
+          {/* <h1 className="">404</h1> */}
+          <img width="100%" style={{ maxWidth: "300px" }} src={pencilgif} />
```
Comment (payalkherajani) [inline src/Components/ErrorPage/ErrorPage.jsx]: inline styling

File: src/Components/Header/SearchBar.jsx [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+    }
+  };
+  useEffect(() => {
+    document.addEventListener("click", closeSearchBar);
+
+    return () => {
+      document.removeEventListener("click", closeSearchBar);
+    };
+  }, []);
+
```
Comment (payalkherajani) [inline src/Components/Header/SearchBar.jsx:40]: I am not sure of this here, why do you require vanilla js to close searchbar ?

File: src/utils/server-request.js [hunk 12]
```diff
@@ -0,0 +1,243 @@
...
+    });
+
+    if (status === 200 || status === 201) {
+      dispatch({
+        type: "SET_WISHLIST",
+        payload: response
+      });
+    }
+  } catch (error) {
+    console.error(error);
```
Comment (payalkherajani) [inline src/utils/server-request.js:50]: remove console

File: src/Components/Header/SearchBar.jsx (hunk 11 above)
Comment (AnkitaBagale) [reply src/Components/Header/SearchBar.jsx:40]: I m trying to add on click on anything in the document except for the search bar should close the search bar. 
I found this way, if there is any other way to do it in react then please let me know.

File: src/App.js (hunk 4 above)
Comment (AnkitaBagale) [reply src/App.js]: Done

File: src/Components/Authentication/ForgotPasswordPage.jsx (hunk 5 above)
Comment (AnkitaBagale) [reply src/Components/Authentication/ForgotPasswordPage.jsx]: Done

File: src/Components/ErrorPage/ErrorPage.jsx (hunk 10 above)
Comment (AnkitaBagale) [reply src/Components/ErrorPage/ErrorPage.jsx]: Done

File: src/utils/server-request.js (hunk 12 above)
Comment (AnkitaBagale) [reply src/utils/server-request.js:50]: Sure, I will remove all the console statements once I implement toast functionality, before deploying to production.

**Pull request merged**